llrpConnector/src/main/java/io/vantiq/extsrc/llrpConnector/LLRPConnector.java

```diff
@@ -0,0 +1,1546 @@
+                            if (connection != null && !connection.isClosed())
+                                connection.close();
+                            return;
+                        }
+
+                        // May want to get the Reader supported version and set to an appropriate version
+
+                        // Get the Reader Capabilities Response
+                        readerCapabilities = getReaderCapabilities();
+
+                        // Create/Send the Reader Configuration and Get the Response
+                        readerConfiguration = sendReaderConfiguration();
+
+                        // Create/Send ROSpec to start reading the tags
+                        sendROSpec(tagReadInterval);
+
+                        // Indicate success
+                        changeReaderStatus(READER_ONLINE);
+
+                        // Send an ENABLE_EVENTS_AND_REPORTS Message to start requesting tag data
+                        ENABLE_EVENTS_AND_REPORTS report = new ENABLE_EVENTS_AND_REPORTS();
+                        write(report, "ENABLE_EVENTS_AND_REPORTS");
+
+                    } catch (VantiqLLRPException | IOException e) {
+                        systemPrint("Unable to connect and startup the reader.");
+                        sendLogMessage(WARN, "connectAndSetupReader: Unable to connect and startup the reader, retrying");
+                        if (connection != null && !connection.isClosed())
+                            try {
+                                connection.close();
+                            } catch (Exception e1) {
+                                systemPrint("Unable to close open connection. ");
+                                log.error("Unable to close open connection. ", e);
+                                sendLogMessage(ERROR, "connectAndSetupReader: Unable to close open connection.\n"
+                                        + ExceptionUtils.getStackTrace(e1));
+                            }
+                    } catch (Exception e) {
+                        systemPrint("Unexpected error while attempting to connect to the reader");
+                        log.error("Unexpected error while attempting to connect to the reader ", e);
+                        sendLogMessage(ERROR, "connectAndSetupReader: Unexpected error.\n "
+                                + ExceptionUtils.getStackTrace(e));
+                    }
+                } else {
+                    systemPrint("connectAndSetupReader: reader is offline");
+                    sendLogMessage(ERROR, "connectAndSetupReader: reader is offline");
+                }
+            } catch (Exception e) {
+                systemPrint("connectAndSetupReader: Unexpected error");
+                log.error("connectAndSetupReader: Unexpected error ", e);
+                sendLogMessage(ERROR, "connectAndSetupReader: Unexpected error.\n" + ExceptionUtils.getStackTrace(e));
+            }
+        });
+        setup.start();
```

Comment:
comment here would be good that this is starting the thread.  Even better might be to encapsulate the thread code into a method so its easier to see the structure.

Reply:
Changed this method to remove the thread declaration and start.  Moved the structure of calling this method in a thread to the calling method:

        // Connect and set up the Reader in a thread - attempt every minute
        Thread setup = new Thread(() -> {
            connectAndSetupReader();
        });
        // Start up the connection/setup thread
        setup.start();